src/Microsoft.Kusto.ServiceLayer/Program.cs

```diff
@@ -60,5 +69,20 @@ internal static void Main(string[] args)
                 Logger.Close();
             }
         }
+
+        private static void CheckParentStatusLoop(Process parent)
```

Comment:
I'd recommend building the logic directly in the Hosting dll to avoid duplicate code.